ophyd/areadetector/cam.py

```diff
@@ -46,7 +46,9 @@ def __init__(self, *args, **kwargs):
 
     def ensure_nonblocking(self):
         self.stage_sigs['wait_for_plugins'] = 'Yes'
-        for c in self.component_names:
+        for c in self.parent.component_names:
+            if c is self:
```

Comment:
`c` here is a string so this will never be true.

Reply:
force-pushed